### PR TITLE
modify DockerMachie Api version to v1alpha2

### LINKED ISSUE #143: expected API version is v1alpha2 not v1alpha1
/kind bug

All types created in this provider should be of type v1alpha2 in order to minimize user confusion.

cc @ericzhang08

### PR DESCRIPTION
**Pull request by ericzhang08**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #143 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: controllers/dockermachine_controller.go [hunk 1]
```diff
@@ -59,12 +61,36 @@ func (r *DockerMachineReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
...
 func (r *DockerMachineReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&infrastructurev1alpha1.DockerMachine{}).
+		For(&infrastructurev1alpha2.DockerMachine{}).
 		Watches(
-			&source.Kind{Type: &infrastructurev1alpha1.DockerMachine{}},
+			&source.Kind{Type: &infrastructurev1alpha2.DockerMachine{}},
 			&handler.EnqueueRequestsFromMapFunc{
-				ToRequests: util.MachineToInfrastructureMapFunc(capiv1alpha2.SchemeGroupVersion.WithKind("Machine")),
+				ToRequests: MachineToDockerMachine(capiv1alpha2.SchemeGroupVersion.WithKind("Machine")),
```
Comment (detiber) [inline controllers/dockermachine_controller.go]: This should likely be reverted back to using `util.MachineToDockerMachine()`

File: controllers/dockermachine_controller.go [hunk 2]
```diff
@@ -59,12 +61,36 @@ func (r *DockerMachineReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
...
 			},
 		).
 		Complete(r)
 }
+
+// MachineToDockerMachine watches v1alpha2 machines and enqueues associated DockerMachines
+func MachineToDockerMachine(gvk schema.GroupVersionKind) handler.ToRequestsFunc {
```
Comment (detiber) [inline controllers/dockermachine_controller.go]: This should be removed with the reversion to `util.MachineToDockerMachine()` above.

File: controllers/dockermachine_controller.go (hunk 1 above)
Comment (chuckha) [reply controllers/dockermachine_controller.go]: ah yeah, looks like the merge conflict got resolved in the wrong order

**Comment by chuckha** [comment]
@detiber I'm ok merging this and fixing later on, there is still much work to do here

**Comment by detiber** [comment]
@chuckha works for me as long as we track the work to followup

**Comment by chuckha** [comment]
@detiber done ! https://github.com/kubernetes-sigs/cluster-api-provider-docker/issues/146

**Pull request merged**